### PR TITLE
Add steam-rom-manager

### PR DESCRIPTION
**Pull request by Witherking25**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:https://github.com/SteamGridDB/steam-rom-manager/issues/287**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge). **The upstream author owns it**
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Witherking25** [comment]
bot, build com.steamgriddb.steam-rom-manager

**Comment by Witherking25** [comment]
bot, build com.steamgriddb.steam-rom-manager

File: com.steamgriddb.steam-rom-manager.json [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+					"sha256": "dfcef88c8e8b262f548c8d2799ff1f13e688bd0f710b5afdf780f9fc2686243b"
+				},
+				{
+					"type": "file",
+					"dest-filename": "srm_512.png",
+					"url": "https://github.com/SteamGridDB/steam-rom-manager/raw/master/src/assets/icons/512x512.png",
+					"sha256": "371dbd9c9dbee38018ad32cff823660c91407f6b23ac7eec7d8860f5f271daaa"
+				},
+				{
+					"type": "extra-data",
```
Comment (hfiguiere) [inline com.steamgriddb.steam-rom-manager.json:117]: since this is GPL licensed software, do you really need to make this extra-data?

Comment (Witherking25) [reply com.steamgriddb.steam-rom-manager.json:117]: technically no

Comment (Witherking25) [reply com.steamgriddb.steam-rom-manager.json:117]: i can make it not extra if you like :)

**Comment by Witherking25** [comment]
bot, build com.steamgriddb.steam-rom-manager

**Comment by Witherking25** [comment]
bot, build com.steamgriddb.steam-rom-manager

**Comment by Witherking25** [comment]
bot, build com.steamgriddb.steam-rom-manager

**Comment by Witherking25** [comment]
ready

**Comment by Witherking25** [comment]
bot, build com.steamgriddb.steam-rom-manager

**Comment by Witherking25** [comment]
bot, build com.steamgriddb.steam-rom-manager

File: com.steamgriddb.steam-rom-manager.json [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	"command": "steam-rom-manager",
+	"separate-locales": false,
+	"finish-args": [
+		"--share=ipc",
+		"--socket=x11",
+		"--socket=pulseaudio",
+		"--share=network",
+		"--device=all",
+		"--filesystem=host",
+		"--allow=devel",
```
Comment (barthalion) [inline com.steamgriddb.steam-rom-manager.json:18]: What is `--allow-devel` needed for?

**Comment by barthalion** [comment]
I think the app should detect whether the user has installed Steam directly on the host or use flatpaked Steam with files at `~/.var/app/com.valvesoftware.Steam `. I know it's configurable, but it's very likely flatpak users use flatpaked Steam.

cc @doZennn

**Comment by Witherking25** [comment]
Flatpak steam is currently UNSUPPORTED in the upstream, so unfortunately no. I can’t support it here

File: com.steamgriddb.steam-rom-manager.json (hunk 2 above)
Comment (Witherking25) [reply com.steamgriddb.steam-rom-manager.json:18]: Testing sorry

**Comment by Witherking25** [comment]
bot, build com.steamgriddb.steam-rom-manager

**Comment by Witherking25** [comment]
bot, build com.steamgriddb.steam-rom-manager

File: com.steamgriddb.steam-rom-manager.json [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+			"buildsystem": "simple",
+			"build-commands": [
+				"install -D run.sh /app/bin/steam-rom-manager",
+				"cp /usr/bin/ar /app/bin",
+				"ar x srm.deb",
+				"tar xf data.tar.xz",
+				"mv \"opt/Steam ROM Manager\" \"/app/srm\"",
+				"rm -rf \"usr/share/icons/hicolor/1024x1024\"",
+				"cd usr; find share/icons/hicolor -type f -exec install -Dm644 \"{}\" \"/app/{}\" \\;",
+				"find /app/share/icons/hicolor -name \'*.png\' -type f -exec bash -c \'mv \"$1\" \"${1/\\/steam-rom-manager/\\/com.steamgriddb.steam-rom-manager}\"\' -- {} \\;",
```
Comment (hfiguiere) [inline com.steamgriddb.steam-rom-manager.json:49]: you can just use `rename-icon: steam-rom-manager` (top-level of the manifest)

Comment (Witherking25) [reply com.steamgriddb.steam-rom-manager.json:49]: 🤦‍♂️

Comment (Witherking25) [reply com.steamgriddb.steam-rom-manager.json:49]: I spent like 2 hours trying to figure out how to do that lol

**Comment by Witherking25** [comment]
bot, build com.steamgriddb.steam-rom-manager

**Comment by doZennn** [comment]
Can @cbartondock and I be invited to be a collaborator too?

**Comment by Witherking25** [comment]
I can try

**Comment by Witherking25** [comment]
Don’t have access to the repository settings so I can’t add you. An admin might need to do that

**Comment by Witherking25** [comment]
For now if you need to change anything do a pr

**Pull request closed without merging**